src/main/java/cd/go/plugin/config/yaml/YamlFileParser.java

```diff
@@ -22,10 +23,14 @@ public JsonConfigCollection parseFiles(File baseDir, String[] files) {
             InputStream inputStream = null;
             try {
                 inputStream = new FileInputStream(new File(baseDir, file));
-                YamlReader reader = new YamlReader(new InputStreamReader(inputStream));
+                YamlConfig config = new YamlConfig();
+                config.setAllowDuplicates(false);
+                YamlReader reader = new YamlReader(new InputStreamReader(inputStream), config);
                 Object rootObject = reader.read();
                 JsonConfigCollection filePart = rootTransform.transform(rootObject, file);
                 collection.append(filePart);
+            } catch (YamlReader.YamlReaderException ex) {
+                collection.addError("YAML contains duplicate keys: " + ex.getMessage() , file);
```

Comment:
I presume that `YamlReader.YamlReaderException` is thrown also in other cases, when there is anything else wrong with yaml format, isn't it?
That catch would wrongly report such case as duplicate key, misleading user to think that there is some duplicate, while actually something else failed in parser.
Can you somehow identify that exception cause was actually a duplicate? Even checking the message in `ex` to contain `"duplicate"` would be fine.
Ideally there should be another test case with invalid yaml, which is not a duplicate, asserting sensible error message.